test/tools/yulopti.cpp

```diff
@@ -87,37 +89,70 @@ class YulOpti
 		}.printErrorInformation(_errors);
 	}
 
-	void parse(string const& _input)
+	shared_ptr<Object> getSubObject(shared_ptr<Object> const& _object, string const& _qualifiedPath)
+	{
+		if (_qualifiedPath.empty() || _qualifiedPath == _object->name.str())
+			return _object;
+
+		yulAssert(
```

Comment:
I don't think the usage of `yulAssert` or any assert is appropriate here. `_qualifiedPath` is user provided and the user may make a typo, which is a perfectly valid scenario. Asserts should not be used for error handling. You could instead throw an exception here (although I'd argue that the user providing and incorrect object path is not exactly exceptional either), or even better, return a `nullptr` indicating that the object was not found, and then handle appropriately elsewhere. Same thing goes for the assert on line 110. The one on line 117 seems OK though.